src/otx/engine/engine.py

```diff
@@ -449,16 +451,22 @@ def explain(
 
         self._build_trainer(**kwargs)
 
-        self.trainer.predict(
+        predictions = self.trainer.predict(
             model=lit_module,
             datamodule=datamodule,
             ckpt_path=ckpt_path,
         )
+
         # Optimize for memory <- TODO(negvet)
-        saliency_maps = self.trainer.model.model.explain_hook.records
+        raw_saliency_maps = self.trainer.model.model.explain_hook.records
         # Temporary saving saliency map for image 0, class 0 (for tests)
-        cv2.imwrite(str(Path(self.work_dir) / "saliency_map.tiff"), saliency_maps[0][0])
-        return saliency_maps
+        cv2.imwrite(str(Path(self.work_dir) / "saliency_map.tiff"), raw_saliency_maps[0][0])
+
+        return get_processed_saliency_maps(
+            raw_saliency_maps,
+            explain_config,
+            predictions,
+        )
```

Comment:
It's better to put `cv2.imwrite` after processing in `get_processed_saliency_maps`, even if it's dummy now

Reply:
Fixed